.github/workflows/build.yml

```diff
@@ -0,0 +1,111 @@
+name: Publish wheels
+
+on:
+  push:
+    tags:
+      - v*
+  pull_request:
+    paths:
+      - 'Cargo.toml'
+      - 'pyproject.toml'
+
+jobs:
+  build:
+    # This workflow comes from https://github.com/pydantic/pydantic-core/blob/main/.github/workflows/ci.yml
+    name: build on ${{ matrix.os }} (${{ matrix.target }} - ${{ matrix.manylinux }})
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - os: windows
+            ls: dir
+            target: x86_64
+            manylinux: auto
+            interpreter: 3.7 3.8 3.9 3.10 3.11
+          - os: windows
+            ls: dir
+            target: i686
+            manylinux: auto
+            python-architecture: x86
+            interpreter: 3.7 3.8 3.9 3.10 3.11
+          - os: macos
+            target: x86_64
+            manylinux: auto
+          - os: macos
+            target: aarch64
+            manylinux: auto
+            interpreter: 3.7 3.8 3.9 3.10 3.11
+          # Build all manylinux targets on 2_24
+          # https://github.com/pypa/manylinux#readme
+          # https://github.com/PyO3/maturin-action/blob/135c746/src/index.ts#L33
+          - os: ubuntu
+            target: x86_64
+            manylinux: 2_24
+          - os: ubuntu
+            target: aarch64
+            manylinux: 2_24
+          - os: ubuntu
+            target: i686
+            manylinux: 2_24
+          - os: ubuntu
+            target: armv7
+            manylinux: 2_24
+            interpreter: 3.7 3.8 3.9 3.10 3.11
+          - os: ubuntu
+            target: ppc64le
+            manylinux: 2_24
+            interpreter: 3.7 3.8 3.9 3.10 3.11
+          - os: ubuntu
+            target: s390x
+            manylinux: 2_24
+            interpreter: 3.7 3.8 3.9 3.10 3.11
+          # musllinux - https://musl.libc.org/about.html
+          - os: ubuntu
+            target: x86_64
+            manylinux: musllinux_1_1 # /!\ value used in steps conditions below.
+          - os: ubuntu
+            target: aarch64
+            manylinux: musllinux_1_1
+
+    runs-on: ${{ matrix.os }}-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: set up python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.11'
+          architecture: ${{ matrix.python-architecture || 'x64' }}
```

Comment:
Even though it'd be more verbose in the config, since we have so may options there anyway, should we just add `python-architecture` to every `include` entry, so it's obvious what config options go together?